### PR TITLE
add required status checks for PR and master

### PR DESCRIPTION
**Pull request by nadade**

### CODE REVIEW

**Comment by nadade** [comment]
do not merge yet (need to do some more testing)

File: blackbelt/handle_github.py [hunk 1]
```diff
@@ -160,24 +160,20 @@ def merge(pr_url):
...
-    }
-
-    r = requests.get(pr_api_url, headers=headers)
-
-    if (r.status_code != 200):
-        raise ValueError("Cannot retrieve PR info with status code %s: %s" % (r.status_code, r))
-
-    pr = r.json()
-
+    pr = get_json_response("https://api.github.com/repos/%(owner)s/%(name)s/pulls/%(number)s" % pr_info)
```
Comment (abtris) [inline blackbelt/handle_github.py]: not from circleci but from github? ok, that can works too.

**Comment by abtris** [comment]
Looks good to me.

File: blackbelt/handle_github.py (hunk 1 above)
Comment (nadade) [inline blackbelt/handle_github.py]: @abtris this way we make sure all required checks are passing (no conflicts with base, different CIs, security checks, etc.)

Comment (nadade) [inline blackbelt/handle_github.py]: at least that's what I was going for but now I see it's an array of all past statuses; will reiterate on that

**Comment by abtris** [comment]
👍 great for tests

**Comment by honzajavorek** [comment]
I believe this fixes half of https://github.com/apiaryio/black-belt/issues/124.

**Pull request merged**